pipelines/mitx/mfe-frontend-app-learning.yml

```diff
@@ -1,65 +1,68 @@
-              - |
-                echo ACCESS_TOKEN_COOKIE_NAME='edx-jwt-cookie-header-payload' > ./mfe-app-learning/.env
-                echo BASE_URL='https://mitx-qa-next.mitx.mit.edu' >> ./mfe-app-learning/.env
-                echo CSRF_TOKEN_API_PATH='/csrf/api/v1/token' >> ./mfe-app-learning/.env
-                echo LMS_BASE_URL='https://mitx-qa-next.mitx.mit.edu' >> ./mfe-app-learning/.env
-                echo LOGIN_URL='https://mitx-qa-next.mitx.mit.edu/login' >> ./mfe-app-learning/.env
-                echo LOGOUT_URL='https://mitx-qa-next.mitx.mit.edu/logout' >> ./mfe-app-learning/.env
-                echo MARKETING_SITE_BASE_URL='https://mitx-qa-next.mitx.mit.edu' >> ./mfe-app-learning/.env
-                echo REFRESH_ACCESS_TOKEN_ENDPOINT='https://mitx-qa-next.mitx.mit.edu/login_refresh' >> ./mfe-app-learning/.env
-                echo SEARCH_CATALOG_URL='https://mitx-qa-next.mitx.mit.edu/courses' >> ./mfe-app-learning/.env
-                echo SITE_NAME='MITx QA' >> ./mfe-app-learning/.env
-                echo STUDIO_BASE_URL='https://studio-mitx-qa-next.mitx.mit.edu' >> ./mfe-app-learning/.env
-                echo USER_INFO_COOKIE_NAME='edx-user-info' >> ./mfe-app-learning/.env
-                cd ./mfe-app-learning
-                npm install
-                NODE_ENV=production npm run build
-      - put: mfe-app-learning-bucket
-        params:
-          source: mfe-app-learning/dist
-          destination:
-            - dir: s3-remote:mitx-mfe-mitx-qa/app-learning
+- name: build
+  plan:
+  - get: mfe-app-learning
+    trigger: true
+  - task: npm-install-and-build
+    config:
+      platform: linux
+      image_resource:
+        type: registry-image
+        source:
+          repository: node
+      inputs:
+      - name: mfe-app-learning
+      outputs:
+      - name: mfe-app-learning/dist
+      params:
+        ACCESS_TOKEN_COOKIE_NAME: edx-jwt-cookie-header-payload
+        BASE_URL: https://learning-mfe-qa.mitxonline.mit.edu
+        CSRF_TOKEN_API_PATH: /csrf/api/v1/token
+        LMS_BASE_URL: https://lms-qa.mitxonline.mit.edu
+        LOGIN_URL: https://lms-qa.mitxonline.mit.edu/login
+        LOGOUT_URL: https://lms-qa.mitxonline.mit.edu/logout
+        MARKETING_SITE_BASE_URL: https://lms-qa.mitxonline.mit.edu
+        REFRESH_ACCESS_TOKEN_ENDPOINT: https://lms-qa.mitxonline.mit.edu/login_refresh
+        SEARCH_CATALOG_URL: https://lms-qa.mitxonline.mit.edu/courses
+        SITE_NAME: MITx Online QA
+        STUDIO_BASE_URL: https://studio-qa.mitxonline.mit.edu
+        USER_INFO_COOKIE_NAME: edx-user-info
+        SESSION_COOKIE_DOMAIN: .mitxonline.mit.edu
+        NODE_ENV: production
+      run:
+        path: sh
+        dir: mfe-app-learning
+        args:
+        - -exc
+        - |
+          npm install
+          npm run build
+  - put: mfe-app-learning-bucket
+    params:
+      source: mfe-app-learning/dist
+      destination:
+      - dir: s3-remote:mitxonline-edxapp-mfe-qa/learning
```

Comment:
Didn't we conclude that assets have to live at the root of the bucket?

Reply:
Yes, but I'm still hopeful